### PR TITLE
SYS-3360 - adjusted registerValidator interface and tests

### PR DESCRIPTION
**Pull request by ivan-cholakov**
This pr adjusts the interface of the registerValidator function that interfaces with the smart contract. Additionally tests are refactored to run properly and additional util function to decompress eth_public_key
Before you submit a PR, please check that:

- [x] You describe the purpose of the PR, e.g.:
  - What does it do?
  - Highlight what important points reviewers should know about;
  - Indicates if there is something left for follow-up PRs.
- [ ] Documentation updated
- [x] Business logic tested successfully
- [x] Verify First, Write Last: In Substrate development, it is important that you always ensure preconditions are met and return errors at the beginning. After these checks have completed, then you may begin the function's computation.

✄ -----------------------------------------------------------------------------

### CODE REVIEW

File: pallets/validators-manager/src/lib.rs [hunk 1]
```diff
@@ -703,10 +700,30 @@ impl<T: Config> Pallet<T> {
...
+        let decompressed = secp256k1::PublicKey::parse_slice(
+            &compressed_eth_public_key.0,
+            Some(secp256k1::PublicKeyFormat::Compressed),
+        );
+        match decompressed {
+            Ok(public_key) => {
+                let decompressed = public_key.serialize();
+                let mut m = [0u8; 64];
+                m.copy_from_slice(&decompressed[1..65]);
+                println!("{:?}", H512::from_slice(&m));
```
Comment (nahuseyoum) [inline pallets/validators-manager/src/lib.rs:715]: I don't think this will pass compilation

File: pallets/validators-manager/src/tests.rs [hunk 2]
```diff
@@ -69,6 +72,32 @@ fn lydia_test_register_existing_validator() {
     });
 }
 
+#[test]
+fn test_decompress_eth_public_key() {
+    // "021f21d300f707014f718f41c969c054936b7a105a478da74d37ec75fa0f831f87"
```
Comment (alexpArtos) [inline pallets/validators-manager/src/tests.rs:77]: Comment does not match the key in the code below. Should it?

File: pallets/validators-manager/src/tests.rs [hunk 3]
```diff
@@ -671,7 +705,9 @@ mod add_validator {
 
             AddValidatorContext {
                 collator,
-                validator_eth_public_key: ecdsa::Public::unchecked_from([0; 33]),
+                collator_eth_public_key: ecdsa::Public::from_raw(hex!(
+                    "02407b0d9f41148bbe3b6c7d4a62585ae66cc32a707441197fa5453abfebd31d57"
```
Comment (alexpArtos) [inline pallets/validators-manager/src/tests.rs:709]: Could we make this a constant and use it everywhere instead?

**Review by alexpArtos — APPROVED** [review]
:dart:

**Review by nahuseyoum — APPROVED** [review]

File: pallets/validators-manager/src/lib.rs [hunk 4]
```diff
@@ -703,10 +700,29 @@ impl<T: Config> Pallet<T> {
...
+        match decompressed {
+            Ok(public_key) => {
+                let decompressed = public_key.serialize();
+                let mut m = [0u8; 64];
+                m.copy_from_slice(&decompressed[1..65]);
+                Ok(H512::from_slice(&m))
+            },
+            Err(err) => Err(err),
+        }
+    }
```
Comment (thadouk) [inline pallets/validators-manager/src/lib.rs:719]: I think this should be moved to the avn-common crate under primitives and unit-tested there.

File: pallets/validators-manager/src/tests.rs [hunk 5]
```diff
@@ -69,6 +72,24 @@ fn lydia_test_register_existing_validator() {
     });
 }
 
+#[test]
+fn test_decompress_eth_public_key() {
```
Comment (thadouk) [inline pallets/validators-manager/src/tests.rs:76]: ditto

**Pull request merged**